obspy/signal/invsim.py

```diff
@@ -763,6 +764,76 @@ def estimate_wood_anderson_amplitude_using_response(response, amplitude,
     return wa_ampl
 
 
+def _paz_to_freq_resp(freqs, zeros, poles, scale_fac):
+    b, a = scipy.signal.ltisys.zpk2tf(zeros, poles, scale_fac)
+    if not isinstance(a, np.ndarray) and a == 1.0:
+        a = [1.0]
+    return scipy.signal.freqs(b, a, freqs * 2 * np.pi)[1]
+
+
+def _unpack_paz(x, numpoles, numzeros):
+    poles_real = np.array(x[:numpoles * 2:2])
+    poles_imag = np.array(x[1:numpoles * 2:2])
+    poles = poles_real + 1j * poles_imag
+    zeros_real = np.array(x[numzeros * 2:numzeros * 2 + numzeros * 2:2])
+    zeros_imag = np.array(x[numzeros * 2 + 1:numzeros * 2 + numzeros * 2:2])
+    zeros = zeros_real + 1j * zeros_imag
+    scale_fac = x[-1]
+    return poles, zeros, scale_fac
+
+
+def _response_misfit(reference, other):
+    """
+    Calculate a misfit measure from two complex response vectors
+    """
+    # just the simple euclidean distance for now. tweaking this might improve
+    # the fitting maybe..
+    misfit = scipy.spatial.distance.euclidean(reference, other)
```

Comment:
If this converges it is probably hard to create a better misfit functional - if you'll find a few examples where it does not converge one has to think a bit harder. I think that in particular the phase might sometimes be a bit hard to fit.

Reply:
Well like mentioned above (or at least somewhere) I see a tendency of very curvy response results, that matches on the sampling points but doesn't make much sense, while still having a very low misfit. So I think the euclidean might be a good start but I am pretty sure it needs a penalty if getting to curvy in between sampling points.. maybe add a second misfit criterion that compares to a simple linear interpolation of the original sampling points but with a much denser sampling.

Reply:
I think the spurious oscillations could be reduced by just limiting the maximum number of poles and zeros based on the number of initial response curve values. 20 poles and zeros are 80 degrees of freedom which is too much for the given data without some form of regularization. A smoother curve can also be enforced via regularization if so desired.

Reply:
Yeah, I agree, I just wanted to leave it fully unconstrained for now. Maybe it would be easiest to just calculate the misfit not only on sample points but maybe on oversampled points with just linear interpolation (but giving more weight to the actual sampling points)

Reply:
Regularizing can have the same effect and would be less hacky I think. Adding some kind of term to the misfit that promotes smoothness of the solution might work.